src/components/Tabs/Annotation/AnnotationImportButton.tsx

```diff
@@ -0,0 +1,165 @@
+import { UploadOutlined, WarningOutlined } from "@ant-design/icons";
+import { Card, Modal, Upload, UploadFile } from "antd";
+import React, { ReactElement, useContext, useMemo, useState } from "react";
+import styled from "styled-components";
+
+import { AnnotationState } from "../../../colorizer/utils/react_utils";
+import { useViewerStateStore } from "../../../state";
+import { FlexColumn, FlexRow } from "../../../styles/utils";
+
+import { AnnotationData } from "../../../colorizer/AnnotationData";
+import { AppThemeContext } from "../../AppStyle";
+import TextButton from "../../Buttons/TextButton";
+
+type AnnotationImportButtonProps = {
+  annotationState: AnnotationState;
+};
+
+const WarningStyleCard = styled(Card)`
+  border-color: var(--color-alert-warning-border);
+  background-color: var(--color-alert-warning-fill);
+`;
+
+const WarningCard = (props: React.PropsWithChildren): ReactElement => {
+  return (
+    <WarningStyleCard size="small">
+      <FlexRow $gap={10}>
+        <div>
+          <WarningOutlined style={{ color: "var(--color-text-warning)", fontSize: "var(--font-size-label)" }} />
+        </div>
+        {props.children}
+      </FlexRow>
+    </WarningStyleCard>
+  );
+};
+
+export default function AnnotationImportButton(props: AnnotationImportButtonProps): ReactElement {
+  const dataset = useViewerStateStore((state) => state.dataset);
+  const { annotationState } = props;
+  const theme = useContext(AppThemeContext);
+
+  const [uploadedFile, setUploadedFile] = useState<File | null>(null);
+  const [convertedAnnotationData, setConvertedAnnotationData] = useState<AnnotationData | null>(null);
+  const [showModal, setShowModal] = useState(false);
+  const [errorText, setErrorText] = useState("");
+
+  const modalContainerRef = React.useRef<HTMLDivElement>(null);
+
+  const handleFileUpload = async (file: File): Promise<void> => {
```

Comment:
This function does not need to be `async`, even if the `reader.onload` below does